### ISSUE TITLE
process title is too much short

### ISSUE THREAD

**Issue by fluca1978** [agroal/pgagroal#215]
Version 1.4 (`HEAD`):
```
% pgagroal -d       
% ps -auxw | grep pgagr
luca       16998  0.0  0.3  69316  2656 ?        Ss   05:59   0:00 pgagroal: ma
```

the process title should be `pgagroal: main`, according to https://github.com/agroal/pgagroal/blob/master/src/libpgagroal/utils.c#L774 but it is truncated at https://github.com/agroal/pgagroal/blob/master/src/libpgagroal/utils.c#L777  because `max_process_title_size` is lower. The problem appears to be here https://github.com/agroal/pgagroal/blob/master/src/libpgagroal/utils.c#L762 because the `max_process_title` is computed on the basis of `argv` and therefore it depends on the length of the command line.
In other words, invoking `pgagroal` will result in `max_process_title_suze` to be `9`, invoking `pgagroal -d` will make `max_process-title_size` to be 12, and `/usr/local/bin/pgagroal -d` will make `max_process_title_size` bigger.

I don't know if we should care about `argv` at such level, I don't know if there's a way to get a safe length for the process title and then to use such as `max_process_title_size` default value.

**Comment by jesperpedersen** [comment]
Note, that this is a PITA area to deal with - see the numerous updates to the function in question.

Make sure that you verify any patches on a harden OS like RHEL 8 or Rocky8 - as they are likely to give you a different result; especially if the string is larger than `argv`.

**Comment by fluca1978** [comment]
I'm not expert in this area, but as far as I can tell,  there's no way to update the process title in Linux without overwriting environment, while other platforms provide functions (.e.g, BSD `setproctitle`). I suspect that you are referring to the fact that the hardened systems will complain about a process title changing over its initial size.
I will investigate on this as soon as I've time.

**Comment by jesperpedersen** [comment]
Yeah, having `setproctitle` on Linux would be much easier... Don't break too many keyboards on this though

**Comment by fluca1978** [comment]
Seems however to work on Rocky Linux 8.5. I'm trying to figure out when it is not working.
